k2/addons/k2-serial/zserial/connection.py

```diff
@@ -27,29 +27,48 @@ class SerialConnectionError(Exception):
 
 class SerialConnectionWrapper(object):
 
-    def __init__(self, thread, connection):
+    def __init__(self, thread, connection, serial, protocol_factory):
         self.thread = thread
         self.connection = connection
+        self.serial = serial
+        self.protocol_factory = protocol_factory
 
     def close(self):
         self.thread.close()
 
+    def suspend(self):
+        self.thread.stop()
```

Comment:
Why use .stop() instead of .close()?
Close seems to do more things and at first glans looks more appropriate since we overwriting the thread-instance when we resume.

Reply:
Never mind, reading the documentation of ReaderThread a bit more careful, it seems the only way is to use .stop() and create a new instance. So ignore above comment.

Reply:
Yes, I wanted to keep the serial connection open so as to not loose any potential data, and "restarting" a thread in python is a no-go.